roles/vagrant_workarounds/tasks/main.yml

```diff
@@ -13,3 +13,24 @@
   shell: rm -f /boot/*-generic.img
   when:
     - ansible_os_family == 'Debian'
+
+# workaround for https://github.com/puppetlabs/puppet/issues/9268
+- name: never install Puppet 7.29.0 on Debian
+  copy:
+    dest: /etc/apt/preferences.d/puppet.pref
+    content: |
+      Package: puppet-agent
+      Pin: version 7.29.0*
+      Pin-Priority: -10
+  when:
+    - ansible_os_family == 'Debian'
+
+# workaround for https://github.com/puppetlabs/puppet/issues/9268
+- name: never install Puppet 7.29.0 on EL
+  community.general.ini_file:
+    path: /etc/dnf/dnf.conf
```

Comment:
we place it into the main `dnf.conf` as when this role runs `/etc/yum.repos.d/puppet7-release.repo` doesn't exist yet, plus, even if it'd exist, there is a chance that it gets overwritten during one of our "upgrade packages" steps and we really do not want that.